mttn/src/syscall.rs

```diff
@@ -1,3 +1,112 @@
 //! Syscall models for mttn.
 //!
 //! These are only used in "Tiny86" tracing mode.
+
+use crate::trace::Step;
+use crate::trace::{DecreeSyscall, MemoryHint, RegisterFile, SyscallState, Tracee};
+use anyhow::{anyhow, Result};
+
+pub trait SyscallDFA {
+    const POINTER_TRANSITION_BYTES: u32 = 8; // number of pointer bytes offset per syscall transition
+    const DATA_TRANSITION_BYTES: u32 = 8; // number of data bytes consumed per syscall transition
+    fn transition(&self, syscall: DecreeSyscall, ebx: u32, ecx: u32, edx: u32)
+        -> Result<Vec<Step>>;
+}
+
+impl<'a> SyscallDFA for Tracee<'a> {
+    fn transition(
+        &self,
+        syscall: DecreeSyscall,
+        ebx: u32, // NOTE(jl): non-`mut`; constant FD for currently supported syscalls.
+        mut ecx: u32,
+        mut edx: u32,
+    ) -> Result<Vec<Step>> {
+        match syscall {
```

Comment:
Long term, we may want to reconsider whether we should translate to the DecreeSyscall model as in trace.rs, but this should be fine for now.